rbs/CommentsAssociator.cc

```diff
@@ -86,191 +227,388 @@ void CommentsAssociator::walkNodes(parser::Node *node) {
     typecase(
         node,
 
-        // Nodes that need to consume preceding comments as well as comments until the end of the node
-
-        [&](parser::Kwbegin *kwbegin) {
+        [&](parser::And *and_) {
```

Comment:
I sorted these alphabetically as we cover a lot of cases. Order is not consistent among the rewriters.